### PR TITLE
Wrap history

### PR DESCRIPTION
**Pull request by generatorpoint**
### What does this PR do?
- Add transaction history component to wrap detail page
- updates some wrap hooks to mutate on tx finish
- changes "wrap/unwrap" button text to "upgrade/downgrade"

note: will make a ticket to do a more thorough change to the wrap/upgrade text

Screenshot 2023-08-10 at 2 50 32 PM

### CODE REVIEW

**Review by iceanddust — APPROVED** [review]
Code looks good as well as the deployed component, good job. I left a comment and found a bug but I think it may have been introduced in a previous PR as I see the same behavior in the Save page, so I think we can merge this one and work on it on a separate PR.

The bug:
When you haven't accepted the T&C and your wallet is not connected, try doing the whole process of accepting and connecting the wallet in the Wrap page, seems like the wrap history component keeps hanging forever (same behavior in the Save page).
To reproduce, clear your local storage and reload the site

File: apps/webapp/src/modules/wrap/components/WrapHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        {wrapHistory ? (
+          <>
+            <TableContainer>
+              <Table variant="simple">
+                <Tbody>
+                  {(!address || (address && !rowsToShow.length)) && (
+                    <Tr>
+                      <Td sx={{ px: 0, pb: 0 }}>
+                        <Text>
+                          {!address && <Trans>Please connect your wallet to see your history</Trans>}
```
[the remark this answers is not in the document]
Comment (bitsforpeace) [reply apps/webapp/src/modules/wrap/components/WrapHistory.tsx:62]: I think the issue is that when there is no account connected, it defaults to mainnet, and so it's querying the mainnet subgraph URL, but that one hasn't been updated with the new queries for wrap history. So the hook is returning an error but we're not handling it yet.

Screenshot 2023-08-11 at 9 52 32 AM

File: apps/webapp/src/modules/wrap/components/WrapHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                          {address && !rowsToShow.length && <Trans>No transactions found</Trans>}
+                        </Text>
+                      </Td>
+                    </Tr>
+                  )}
+                  {rowsToShow.map(row => (
+                    <Tr key={row.transactionHash}>
+                      {/*todo: add in trans macros to here and etherscan link*/}
+                      <Td border="none" w="34%">
+                        {(getWrapAmount(row) > 0 ? t`Upgraded` : t`Downgraded`) +
```
Comment (codewiththeflowxyz) [inline apps/webapp/src/modules/wrap/components/WrapHistory.tsx:72]: I think the t macro here should wrap the whole sentence, it may be other languages that put numbers first in a sentence or something like that.

File: packages/hooks/src/wrap/useWrapHistory.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  const ngtToMkr: MkrNgtRow[] = response.ngtToMkrs.map((w: WrapResponse<MkrNgtRow>) => ({
+    mkrAmt: -BigInt(w.mkrAmt), //make withdrawals negative
+    ngtAmt: -BigInt(w.ngtAmt),
+    blockTimestamp: new Date(parseInt(w.blockTimestamp) * 1000),
+    transactionHash: w.transactionHash
+  }));
+
+  const combined = [...daiToNst, ...nstToDai, ...mkrToNgt, ...ngtToMkr];
+  return combined.sort((a, b) => b.blockTimestamp.getTime() - a.blockTimestamp.getTime());
```
Comment (codewiththeflowxyz) [inline packages/hooks/src/wrap/useWrapHistory.ts:69]: Is it possible to get this sorted directly from theGraph?

**Review by codewiththeflowxyz — COMMENTED** [review]
Looking good, just left a few comments.

**Review by bitsforpeace — APPROVED** [review]
Seems a little inconsistent where one says what it was upgraded FROM and another row just says upgraded.

Screenshot 2023-08-11 at 2 43 41 PM

That whole column feels a bit wordy to me. I feel like the whole table would be cleaner if it was just something like `342.3 MKR -> 23423.23 NGT` but that's a design decision we can talk about outside of the PR.

File: packages/hooks/src/wrap/useWrapHistory.ts (hunk 3 above)
Comment (generatorpoint) [reply packages/hooks/src/wrap/useWrapHistory.ts:69]: good thought, I'll look into that

File: apps/webapp/src/modules/wrap/components/WrapHistory.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (generatorpoint) [reply apps/webapp/src/modules/wrap/components/WrapHistory.tsx]: hmm the query should be disabled when there is no address available, I'll dig into why that happens

[the remark this answers is not in the document]
Comment (generatorpoint) [reply apps/webapp/src/modules/wrap/components/WrapHistory.tsx]: so with react query `isLoading` returns true when a query is disabled, so we have to use `isInitialLoading`:
> Lazy queries will be in status: 'loading' right from the start because loading means that there is no data yet. This is technically true, however, since we are not currently fetching any data (as the query is not enabled), it also means you likely cannot use this flag to show a loading spinner.

File: packages/hooks/src/wrap/useWrapHistory.ts (hunk 3 above)
Comment (generatorpoint) [reply packages/hooks/src/wrap/useWrapHistory.ts:69]: I see we can sort the results of each  individual query (eg `orderBy: blockTimestamp`), but we'd still have to sort the merged results in the client anyway.

But I think we can combine the results using a custom resolver in our subgraph. We'd have to write that so it conforms to a common interface of each query. I think its a good idea but might be a separate ticket.
